src/core/lib/transport/metadata_batch.cc

```diff
@@ -319,7 +371,9 @@ void grpc_metadata_batch_copy(grpc_metadata_batch* src,
     // there's a duplicate entry for a callout.  However, that can't be
     // the case here, because we would not have been allowed to create
     // a source batch that had that kind of conflict.
-    GPR_ASSERT(error == GRPC_ERROR_NONE);
+    GPR_DEBUG_ASSERT(error == GRPC_ERROR_NONE);
+    if (error) {
```

Comment:
Instead of a branch, the common practice is cast to void `(void)error` instead. But, please use  `GRPC_UNUSED` instead.